mutalyzer/variant.py

```diff
@@ -0,0 +1,270 @@
+"""
+"""
+
+from __future__ import unicode_literals
+
+from Bio.SeqUtils import seq3
+
+from extractor import extractor
+
+from mutalyzer import models
+
+weights = {
+    "subst": extractor.WEIGHT_SUBSTITUTION,
+    "del": extractor.WEIGHT_DELETION,
+    "ins": extractor.WEIGHT_INSERTION,
+    "dup": extractor.WEIGHT_INSERTION,
+    "inv": extractor.WEIGHT_INVERSION,
+    "delins": extractor.WEIGHT_DELETION_INSERTION
+}
+
+class HGVSList(list):
```

Comment:
The current implementation might give some surprises, for example:

    >>> a = HGVSList([1,2,3])
    >>> type(a[1:])
    list

    >>> b = HGVSList([4,5,6])
    >>> type(a + b)
    list

This can of course be fixed by implementing the relevant magic methods, but it already hints that we might not really want to subclass `list`. None of the `list` methods will use any of the methods we override, leading to the unexpected behaviour as above and I think that shows `list` is not really meant to be subclassed.

Is our type really a `list` (in the Python sense)? I think it is better modelled as an object containing a `list`:

    class HGVSList(object):
        def __init__(self):
            self.elems = []

        @property
        def weight(self):
            return ...

It might still make sense to make this object iterable (yielding the list elements) for convenience.

So in summary, I think subclassing `list` might lead to confusion while it doesn't really gain us anything. Composition is the simpler and expected solution.

See also: [composition over inheritance](https://encrypted.google.com/search?hl=en&q=composition%20over%20inheritance)